src/ripple/app/paths/Flow.h

```diff
@@ -58,6 +60,7 @@ flow (PaymentSandbox& view,
     bool defaultPaths,
     bool partialPayment,
     bool ownerPaysTransferFee,
+    bool offerCrossing,
     boost::optional<Quality> const& limitQuality,
     boost::optional<STAmount> const& sendMax,
```

Comment:
Yeah.  This wasn't pretty before, with 3 bools and an optional.  I made it worse by adding another bool

I can see two options: an OR-able enum, or make `flow` a class with an `invoke()` method.

We could replace the four bools with a single OR-able enum.  I'd have to try it to see if it was an improvement.

If we made `flow` a class then we could call setup methods on it prior to the `invoke`.  That would hide some of the messy aspects of the setup, but it would also make it less clear what the essential parameters are.

I'd need to run an experiment to see whether either of these would be a win.  Shall I try the enum experiment and see what it looks like?